### PR TITLE
Remove setx and use WM_SETTINGCHANGE in install script

### PR DESCRIPTION
**Pull request by Haard30**
Bug fix for https://github.com/AzureAD/microsoft-authentication-cli/issues/166

`setx` truncated the `$PATH` when its size exceeded 1024 chars. As we always append azureauth installation, we were able to repro and catch the bug with install script using `setx`.

As a solution, I am removing the use of `setx` and using WM_SETTINGCHANGE to notify about environment variable changes.

Testing:
Test 1:
1. Remove any AzureAuth installations.
2. Put random values in PATH env var to increase size to 1024. 
3. Run modified install script in Powershell and verify that AzureAuth new version is appended to the PATH and not truncated.

Test2:
1. Follow first 2 steps of Test1.
2. Open `cmd.exe`. Run the install script.
3. Open new cmd.exe and verify that all new processes are notified of the env var change and that azureauth is usable.

### CODE REVIEW

File: install/install.ps1 [hunk 1]
```diff
@@ -194,12 +206,10 @@ function Install-Post-0-4-0 {
                 $newPath = "${currentPath};${targetDirectory}"
             }
         }
-        # Update the $PATH environment variable with any modifications made above. We use `setx` here instead of
-        # directly writing back the registry value because `setx` signals new processes to pick up on the environment
-        # variable changes.
-        setx PATH $newPath > $null
+        Set-ItemProperty -Path $registryPath -Name PATH -Value $newPath
+        Send-SettingChange
```
Comment (reillysiemens) [inline install/install.ps1:210]: Should we use `Set-ItemProperty` or `[Environment]::SetEnvironmentVariable`? Do you know if there's an important difference there, @Haard30?

Comment (Haard30) [reply install/install.ps1:210]: I found using the `Set-ItemProperty` well documented [here](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/set-itemproperty?view=powershell-7.2). I am not aware of any important difference but `[Environment]::SetEnvironmentVariable` directly uses .NET class `[Environment]` and its function.

Comment (kyle-rader) [reply install/install.ps1:210]: Those sound different, `[Enviornment]` using Dotnet's `Enviornment` under the hood implies only setting the env var for this process and not updating the Registry.

**Comment by kyle-rader** [comment]
I think we just need to always run the pipeline, and unfortunately that disables external contributions directly through this PR experience. We could still take them, just logiscally, we would have to manually pull a branch from a fork, and reopen the PR directly.

File: install/install.ps1 (hunk 1 above)
Comment (Haard30) [reply install/install.ps1:210]: I found this doc on `[Environment]` [http://xahlee.info/powershell/environment_variables.html](http://xahlee.info/powershell/environment_variables.html)

**Review by kyle-rader — APPROVED** [review]

**Review by reillysiemens — APPROVED** [review]

**Pull request merged**